lib/ClangImporter/ClangModuleDependencyScanner.cpp

```diff
@@ -204,14 +204,29 @@ static void recordModuleDependencies(
     for (const auto &fileDep : clangModuleDep.FileDeps) {
       fileDeps.push_back(fileDep.getKey().str());
     }
-
+    std::vector<std::string> allArgs = Impl.ClangArgs;
+    for(auto arg: clangModuleDep.NonPathCommandLine) {
+      allArgs.push_back("-Xclang");
+      allArgs.push_back(arg);
+    }
+    ClangImporterOptions Opts;
+    std::vector<std::string> cc1Args;
+    createClangInvocation(this, Opts, allArgs, cc1Args);
+    std::vector<std::string> swiftArgs;
+    swiftArgs.reserve(cc1Args.size() * 3);
+    for (auto arg: cc1Args) {
+      swiftArgs.push_back("-Xcc");
+      swiftArgs.push_back("-Xclang");
```

Comment:
I'm not sure how `-Xcc` works in the Swift driver, but given this is a `-cc1` command line you can't use `-Xclang`, as that's for forwarding driver options to the frontend.